src/manybody_gal.F

```diff
@@ -312,18 +323,9 @@ FUNCTION angular(gal, r_last_update_pbc, iparticle, cell, particle_set, nvec, en
       ENDDO
 
       ! Abort if the oxygen is not part of a water molecule (2 H)
-      IF (count_h == 0) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("No H atoms for O found")
-      ELSEIF (count_h == 1) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Only one H atom of O atom found")
-      ELSEIF (count_h == 3) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Three H atoms for O atom found")
-      ELSEIF (count_h > 3) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Error in gal part: too many H atoms for O")
+      IF (count_h /= 2) THEN
+         CALL cp_abort(__LOCATION__, &
+                 " Error: Found "//cp_to_string(count_h)//" H atoms for O atom "//cp_to_string(iparticle))
       ENDIF
 
       IF (count_h == 2) THEN
```

Comment:
One more nit: This could be an `ELSE` branch.

```
IF (count_h /= 2) THEN
   CALL cp_abort(__LOCATION__, ...
ELSE 
   rih1(:) = ...
   ...
END IF
```

The same applies to `angular_d` below.

Reply:
Yes, I corrected both.
Thanks, I went to quick